### PR TITLE
[New] Add `prop-name-casing`

### PR DESCRIPTION
**Pull request by bfkbx960**
This PR implements rule proposed in #252

### CODE REVIEW

**Comment by michalsnik** [comment]
Thanks for your work @bfkbx960 ! We really appreciate it :) We're going to merge new rules once stable v4 is out 🚀

File: lib/rules/prop-name-casing.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      if (item.type !== 'Property') {
+        return
+      }
+
+      const propName = item.key.type === 'Literal' ? item.key.value : item.key.name
+      const convertedName = converter(propName)
+      if (convertedName !== propName) {
+        context.report({
+          node: item,
+          message: 'Prop "{{name}}" is not {{caseType}}.',
```
Comment (michalsnik) [inline lib/rules/prop-name-casing.js]: I think it should be: `is not in`

File: lib/rules/prop-name-casing.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'enforce specific casing for the Prop name in Vue components',
+      category: 'strongly-recommended'
```
Comment (michalsnik) [inline lib/rules/prop-name-casing.js]: Can you please change the category to be `upcoming`?
```
category: 'upcoming' // 'strongly-recommended'
```
We're not going to enable this rule by default until the next major release, so we'll stash them in special category for everyone to see what's going to be enabled in next version :)

**Comment by bfkbx960** [comment]
@michalsnik suggestions applied, thank you for code review.
In my local environment the test will succeed but it seems to fail here :(

**Comment by michalsnik** [comment]
Thanks @bfkbx960! If you remove `node_modules`, set node to version 4, install all dependencies once again and run `npm test` on node 4 - it's failing. It's working on master though.

**Comment by michalsnik** [comment]
or it's not working even on master now o.O which is soo weird, as yeasterday I released 4.1.0 and everything passed

**Pull request merged**